gravitee-apim-console-webui/src/management/settings/api-portal-header/migrated/api-portal-header.component.spec.ts

```diff
@@ -0,0 +1,232 @@
+import { of } from 'rxjs';
+import { GioConfirmDialogHarness } from '@gravitee/ui-particles-angular';
+
+
+import { ApiPortalHeaderModule } from './api-portal-header.module';
+import { ApiPortalHeaderComponent } from './api-portal-header.component';
+import { ApiPortalHeaderHarness } from './api-portal-header.harness.';
+import { HeaderDialogHarness } from './header-dialog/header-dialog.harness';
+
+import { CONSTANTS_TESTING, GioHttpTestingModule } from '../../../../shared/testing';
+import { EnvironmentApiHeadersService } from '../../../../services-ngx/environment-api-headers.service';
+import { SnackBarService } from '../../../../services-ngx/snack-bar.service';
+import { GioTestingPermissionProvider } from '../../../../shared/components/gio-permission/gio-permission.service';
+import PortalSettingsService from '../../../../services/portalSettings.service';
+
+const headersList = [
+  {
+    id: '8ea',
+    name: 'api.version',
+    value: '${api.version}',
+    order: 1
+  },
+  {
+    id: '084',
+    name: 'api.owner',
+    value: '${api.owner}',
+    order: 2
+  },
+  {
+    id: 'a37',
+    name: 'api.publishedAt',
+    value: '${(api.deployedAt?date)!}',
+    order: 3
+  }
+];
+
+describe('ApiPortalHeaderComponent', () => {
+  const fakeSnackBarService = {
+    success: jest.fn(),
+    error: jest.fn()
+  };
+  const fakeEnvironmentApiHeadersService = {
+    getHeadersList$: () => of(headersList),
+    getApiHeaders: jest.fn(() => of([])),
+    deleteApiHeader: jest.fn(),
+    updateApiHeader: jest.fn(() => of([]))
+  };
+
+  const fakePortalSettingsService = {
+    save: jest.fn(),
+    get: jest.fn()
+  };
+
+  const settings = {
+    portal: {
+      apis:
+        {
+          apiHeaderShowTags: {
+            'enabled': true
+          },
+          apiHeaderShowCategories: {
+            'enabled': true
+          },
+          promotedApiMode: {
+            'enabled': true
+          },
+          tilesMode: {
+            'enabled': true
+          },
+          categoryMode: {
+            'enabled': true
+          }
+        }
+
+    }
+  };
+  let fixture: ComponentFixture<ApiPortalHeaderComponent>;
+  let componentHarness: ApiPortalHeaderHarness;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [ApiPortalHeaderComponent],
+      imports: [
+        GioHttpTestingModule,
+        ApiPortalHeaderModule,
+        BrowserAnimationsModule,
+        NoopAnimationsModule
+      ],
+      providers: [
+        {
+          provide: GioTestingPermissionProvider,
+          useValue: ['environment-settings-c', 'environment-settings-u', 'environment-settings-d']
+        },
+        {
+          provide: 'Constants',
+          useValue: CONSTANTS_TESTING
+        },
+        { provide: MatDialogRef, useValue: {} },
+        { provide: MAT_DIALOG_DATA, useValue: [] },
+        { provide: SnackBarService, useValue: fakeSnackBarService },
```

Comment:
What do you think about avoiding mocking services and expecting and flushing http requests directly?

See the other tests for examples of expect on the httpTestingController 

You can use this test as an example :
https://github.com/gravitee-io/gravitee-api-management/blob/1d708fb6b4456b1adcd2da2e57a4932a20c986bb/gravitee-apim-console-webui/src/management/settings/api-logging/api-logging.component.spec.ts#L30

Reply:
we avoid mocks because they are not typed and do not evolve with the services. so when they change their contract these tests will continue to work whereas the real functionality will not.

Reply:
mocked removed